packages/server/app/Controllers/Http/MpeRoomsHttpController.ts

```diff
@@ -1,45 +1,105 @@
+const MPE_ROOMS_SEARCH_LIMIT = 10;
+
 export default class MpeRoomsHttpController {
     //TODO should list private with invitation etc etc and takes an userID
-    public async listAllRooms(): Promise<ListAllMpeRoomsResponseBody> {
-        const allRooms = await MpeRoom.query().preload('creator');
+    public async listAllRooms({
+        request,
+    }: HttpContextContract): Promise<ListAllMpeRoomsResponseBody> {
+        const { searchQuery, page } = ListAllMpeRoomsRequestBody.parse(
+            request.body(),
+        );
 
-        return await fromMpeRoomsToMpeRoomSummaries({
-            mpeRooms: allRooms,
+        const allMpeRoomsPagination = await MpeRoom.query()
+            .where('name', 'ilike', `${searchQuery}%`)
+            .orderBy([
+                {
+                    column: 'mpe_rooms.is_open',
+                    order: 'asc',
+                },
+                // FIXME: need to handle invitations
+                // {
+                //     column: 'invitationID',
+                //     order: 'asc',
+                // },
+                {
+                    column: 'mpe_rooms.uuid',
+                    order: 'asc',
+                },
+            ])
+            .preload('creator')
+            .paginate(page, MPE_ROOMS_SEARCH_LIMIT);
+
+        const totalRoomsToLoad = allMpeRoomsPagination.total;
+        const hasMoreRoomsToLoad = allMpeRoomsPagination.hasMorePages;
+        const formattedMpeRooms = await fromMpeRoomsToMpeRoomSummaries({
+            mpeRooms: allMpeRoomsPagination.all(),
             userID: datatype.uuid(), //TODO this is temporary we need to be refactor during mpe search engine implem
         });
+
+        return {
+            page,
+            data: formattedMpeRooms,
+            hasMore: hasMoreRoomsToLoad,
+            totalEntries: totalRoomsToLoad,
+        };
     }
 
-    public async listAllUserRooms({
+    public async listMyRooms({
         request,
-    }: HttpContextContract): Promise<LibraryMpeRoomSearchResponseBody> {
+    }: HttpContextContract): Promise<MpeSearchMyRoomsResponseBody> {
         const rawBody = request.body();
         //TODO The userID raw in the request body is temporary
         //Later it will be a session cookie to avoid any security issues
-        const { userID } = MpeRoomSearchRequestBody.parse(rawBody);
+        const { userID, searchQuery, page } =
+            MpeSearchMyRoomsRequestBody.parse(rawBody);
 
         const user = await User.findOrFail(userID);
-        await user.load('mpeRooms', (mpeRoomQuery) => {
-            return mpeRoomQuery.preload('creator');
-        });
+        const mpeRoomsPagination = await user
+            .related('mpeRooms')
+            .query()
+            .where('name', 'ilike', `${searchQuery}%`)
+            .orderBy([
+                {
+                    column: 'mpe_rooms.is_open',
+                    order: 'asc',
+                },
+                // FIXME: need to handle invitations
```

Comment:
Do we want the personal user rooms to be ordered by potential invitation ?

Reply:
I checked our Basecamp documentation, and it seems that we don't, indeed.
Do we even want to filter by opening status?

Reply:
Mhm I don't think so